src/models/architectures/cnns/architectures/competitive_multiscale_cnn.py

```diff
@@ -0,0 +1,74 @@
+import torch
+import torch.nn as nn
+
+
+class MaxOut(nn.Module):
+
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, inputs):
+        cur_max = inputs[0]
+        for inp in inputs[1:]:
+            cur_max = torch.max(cur_max, inp)
+        return cur_max
+
+
+class ConvCompetitiveLayer(nn.Module):
+
+    def __init__(self, in_channels, nb_kernels, kernels_sizes):
+        super(ConvCompetitiveLayer, self).__init__()
+
+        self.convs = [nn.Conv2d(in_channels, nb_kernels, k_sz, padding=k_sz//2)
+                      for k_sz in kernels_sizes]
+
+        self.bnorms = [nn.BatchNorm2d(nb_kernels) for _ in kernels_sizes]
```

Comment:
As you use only kernel_sizes length, maybe it could be better to write it like:
`[nn.BatchNorm2d(nb_kernels)]*len(kernel_sizes)`?

Reply:
Wouldn't it duplicate the same BatchNorm2d object multiple times instead of creating separate ones that should be optimised separately?

Reply:
Yes it does creat copies.